AGENTS.md

```diff
@@ -0,0 +1,340 @@
+# Context for coding agents
+
+This file provides guidance to coding agents when working with code in this repository.
+
+## About AlchemyCMS
+
+AlchemyCMS is an open source Rails CMS engine with a flexible, YAML-driven content architecture. It uses a three-tier content model: Pages → Elements → Ingredients, with multi-language support, versioning, and a modern admin interface built with Rails + Web Components.
+
+## Development Commands
+
+### Initial Setup
+
+```bash
+bin/setup
+```
+
+This installs dependencies (Ruby gems + Bun packages) and sets up the dummy app in `spec/dummy/`.
+
+### Running Tests
+
+```bash
+# Run all tests (prepares database + runs RSpec suite)
+bundle exec rake
+
+# Run RSpec only
+bin/rspec
+
+# Run specific test file
+bin/rspec spec/models/alchemy/page_spec.rb
+
+# Prepare test database (required before first test run)
+bundle exec rake alchemy:spec:prepare
+
+# Run JavaScript tests
+bun run test
+```
+
+### Building Frontend Assets
+
+```bash
+# Build all assets (JavaScript, CSS, Handlebars templates, icons)
+bun run build
+
+# Build individual components
+bun run build:js         # Rollup JavaScript bundling
+bun run build:css        # Sass compilation
+bun run handlebars:compile  # Compile Handlebars templates
+bun run build:icons      # Generate icon sprite
+```
+
+### Running the Dummy App
+
+```bash
+# Start the development server (runs spec/dummy/bin/dev)
+bin/start
+```
+
+This starts a Rails server and bun watch task in the dummy application for running a full dev environment.
+
+### Starting the Rails console
+
+```bash
+cd spec/dummy; bin/rails console; cd -
+```
+
+This starts the Rails console for running code in the context of the dummy application.
+
+### Running a one off script
+
+```bash
+cd spec/dummy; bin/rails runner 'puts Some.ruby.code'; cd -
+```
+
+This starts the Rails console for running code in the context of the dummy application.
+
+### Linting
+
+```bash
+# Ruby linting (uses Standard/RuboCop)
+bundle exec standardrb
+
+# JavaScript linting
+bun run eslint
+
+# Prettier linting
+bun run lint
+```
+
+### Code Formatting
+
+```bash
+# Ruby linting (uses Standard/RuboCop)
+bundle exec standardrb --fix
+
+# Prettier formatting
+bun run prettier --write app/javascript/**/*.js
+```
+
+## Architecture Overview
+
+### Content Model: Three-Tier Hierarchy
+
+Alchemy organizes content in a three-tier hierarchy:
+
+**Pages** (`Alchemy::Page`)
+- Top-level content containers organized in a tree structure (using `awesome_nested_set`)
```

Comment:
Maybe a sentence about how the page holds the URL path by which content is adressed.